app/index.js

```diff
@@ -0,0 +1,104 @@
+var camelcase = require('lodash.camelcase');
+var kebabcase = require('lodash.kebabcase');
+
+module.exports = generators.Base.extend({
+  initializing: function () {
+    this.pkg = require('../package.json');
+  },
+
+  prompting: function () {
+    var done = this.async();
+
+    // Have Yeoman greet the user.
+    this.log(yosay(
+      'Welcome to the ' + chalk.red('Babel Library Boilerplate') + ' generator!'
+    ));
+
+    npm.load(function() {
+      var config = gitConfig.sync();
+      var prompts = [{
+        type: 'input',
+        name: 'user',
+        message: 'What is the Github username/organization for this project?',
+        default: npm.whoami(),
+        store: true
+      }, {
+        type: 'input',
+        name: 'repo',
+        message: 'What is the repository/project name?',
+        default: kebabcase(this.appname)
+      }, {
+        type: 'input',
+        name: 'description',
+        message: 'What is a short description for this project?'
+      }, {
+        type: 'input',
+        name: 'author',
+        message: 'Who is the author of this project?',
+        default: config.user.name + ' <' + config.user.email + '>',
+        store: true
+      }, {
+        type: 'input',
+        name: 'variable',
+        message: 'What is the name of this project\'s main variable?',
+        default: camelcase(this.appname)
+      }, {
+        type: 'confirm',
+        name: 'global',
+        message: 'Expose that variable as a browser global?',
+        default: true
+      }];
+
+      this.prompt(prompts, function (props) {
+        this.user = props.user;
+        this.repo = props.repo;
+        this.description = props.description;
+        this.author = props.author;
+        this.global = props.global;
+        this.variable = props.variable;
+        done();
+      }.bind(this));
+    }.bind(this));
+  },
+
+  writing: {
+    app: function () {
+      this.template('babelrc', '.babelrc');
+      this.template('jscsrc', '.jscsrc');
+      this.template('travis.yml', '.travis.yml');
+      this.template('eslintrc', '.eslintrc');
+      this.template('_package.json', 'package.json');
+      this.template('LICENSE', 'LICENSE');
+      this.template('README.md', 'README.md');
+      this.template('CHANGELOG.md', 'CHANGELOG.md');
+      this.template('gitignore', '.gitignore');
+      this.template('gulpfile.js', 'gulpfile.js');
+      mkdirp.sync('src');
+      this.template('src/index.js', 'src/' + this.repo + '.js');
+      mkdirp.sync('test');
+      this.template('test/eslintrc', 'test/.eslintrc');
+      this.template('test/runner.html', 'test/runner.html');
+      mkdirp.sync('test/setup');
+      this.template('test/setup/browserify.js', 'test/setup/browserify.js');
+      this.template('test/setup/node.js', 'test/setup/node.js');
+      this.template('test/setup/setup.js', 'test/setup/setup.js');
+      mkdirp.sync('test/unit');
+      this.template('test/unit/index.js', 'test/unit/' + this.repo + '.js');
```

Comment:
That would work. I just copy-pasted from James' generator. I don't have a strong opinion tbqh. Do you prefer a loop?